src/test/merkle_tests.cpp

```diff
@@ -0,0 +1,154 @@
+#include "test/test_bitcoin.h"
+
+#include "data/merkle_roots.json.h"
+#include "data/merkle_serialization.json.h"
+#include "data/merkle_witness_serialization.json.h"
+#include "data/merkle_path.json.h"
+
+#include <iostream>
+
+#include "utilstrencodings.h"
+#include "version.h"
+
+#include <boost/test/unit_test.hpp>
+#include "json/json_spirit_reader_template.h"
+#include "json/json_spirit_utils.h"
+#include "json/json_spirit_writer_template.h"
+using namespace json_spirit;
+extern Array read_json(const std::string& jsondata);
+
+#include "zcash/IncrementalMerkleTree.hpp"
+
+//#define PRINT_JSON 1
+
+using namespace std;
+
+template<typename T, typename U>
+void expect_test_vector(T& it, const U& expected)
+{
+    CDataStream ss1(SER_NETWORK, PROTOCOL_VERSION);
+    ss1 << expected;
+
+    #ifdef PRINT_JSON
+    std::cout << "\t\"" ;
+    std::cout << HexStr(ss1.begin(), ss1.end()) << "\",\n";
+    #else
+    std::string raw = (it++)->get_str();
```

Comment:
It would be slightly more clear if the `it++` was done in the same line that calls `expect_test_vector()`.